src/QueryReflection/RuntimeConfiguration.php

```diff
@@ -53,6 +57,19 @@ public function errorMode(string $mode): self
         return $this;
     }
 
+    /**
+     * Defines the PDO default fetch mode.
+     * This might be necessary in case you are using `\PDO::ATTR_DEFAULT_FETCH_MODE`.
+     *
+     * @param QueryReflector::FETCH_TYPE_BOTH|QueryReflector::FETCH_TYPE_ASSOC $mode
```

Comment:
to simplify we only allow a default-fetch-mode of BOTH or ASSOC for now.

@xPaw as I understood it, this is sufficient for your use case?

Reply:
Looking at my code base, my actual use case is setting ASSOC by default on the PDO object, and then sometimes using PDO::FETCH_KEY_PAIR, PDO::FETCH_COLUMN, PDO::FETCH_GROUP, PDO::FETCH_UNIQUE on fetch(all) calls.

Myself, I don't ever use FETCH_NUM.

> I have the feeling that using different fetch-modes on a single statement/result is a rather exotic use case..

I think it makes it easier to think about that setting default mode, or using setFetchMode simply sets the current default, if one is not provided in query/fetch methods. So the proper fix would be to make reflection know all the types as if it was BOTH internally, but apply the correct type based on current fetch mode.

That being said, I don't think the type of PDOStatement itself matters that much, you want to know the resulting type of query/fetch.

Reply:
I have an idea and will try it tomorrow. Stay tuned

Reply:
lets see whether this works: https://github.com/staabm/phpstan-dba/pull/268

Reply:
> That being said, I don't think the type of PDOStatement itself matters that much, you want to know the resulting type of query/fetch.

the type on the statement itself, is the one which is used for the iterator. when its not precise phpstan would think the iterator would contain more offsets, as it actually has.

see 

https://github.com/staabm/phpstan-dba/blob/9a3c964c0497b4dde9204d7e2fd6cf9399c11e05/tests/default/data/pdo.php#L14-L25

Reply:
> https://github.com/staabm/phpstan-dba/issues/269

https://github.com/staabm/phpstan-dba/issues/269